research/flamby/fed_isic2019/ditto_deep_mmd/client.py

```diff
@@ -0,0 +1,200 @@
+import argparse
+import os
+from collections import OrderedDict
+from logging import INFO
+from pathlib import Path
+from typing import Dict, Optional, Sequence, Tuple
+
+import flwr as fl
+import torch
+import torch.nn as nn
+from flamby.datasets.fed_isic2019 import BATCH_SIZE, LR, NUM_CLIENTS, Baseline, BaselineLoss
+from flwr.common.logger import log
+from flwr.common.typing import Config
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from fl4health.checkpointing.checkpointer import BestLossTorchCheckpointer
+from fl4health.checkpointing.client_module import ClientCheckpointModule
+from fl4health.clients.deep_mmd_clients.ditto_deep_mmd_client import DittoDeepMmdClient
+from fl4health.utils.losses import LossMeterType
+from fl4health.utils.metrics import BalancedAccuracy, Metric
+from fl4health.utils.random import set_all_random_seeds
+from research.flamby.flamby_data_utils import construct_fedisic_train_val_datasets
+
+FED_ISIC2019_BASELINE_LAYERS: OrderedDict[str, int] = OrderedDict()
+for i in range(16):
+    FED_ISIC2019_BASELINE_LAYERS[f"base_model._blocks.{i}"] = 64
+FED_ISIC2019_BASELINE_LAYERS["base_model._dropout"] = 1280
+
+
+class FedIsic2019DittoClient(DittoDeepMmdClient):
+    def __init__(
+        self,
+        data_path: Path,
+        metrics: Sequence[Metric],
+        device: torch.device,
+        client_number: int,
+        learning_rate: float,
+        lam: float = 0,
+        loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
+        deep_mmd_loss_weight: float = 10,
+        deep_mmd_loss_depth: int = 1,
+        checkpointer: Optional[ClientCheckpointModule] = None,
+    ) -> None:
+        size_feature_extraction_layers = OrderedDict(
+            list(FED_ISIC2019_BASELINE_LAYERS.items())[-1 * deep_mmd_loss_depth :]
+        )
+        super().__init__(
+            data_path=data_path,
+            metrics=metrics,
+            device=device,
+            loss_meter_type=loss_meter_type,
+            checkpointer=checkpointer,
+            lam=lam,
+            deep_mmd_loss_weight=deep_mmd_loss_weight,
+            flatten_feature_extraction_layers={key: True for key in size_feature_extraction_layers},
```

Comment:
Maybe also form this dictionary above the super call as you're doing with `size_feature_extraction_layers`?